examples/skinned_meshlets/source/skinned_meshlets.cpp

```diff
@@ -0,0 +1,628 @@
+				for (auto pair : distinctMaterials) {
+					if (std::end(pair.second) != std::find(std::begin(pair.second), std::end(pair.second), meshIndex)) {
+						break;
+					}
+
+					drawCallData.mMaterialIndex++;
+				}
+
+				auto selection = gvk::make_models_and_meshes_selection(curModel, meshIndex);
+				std::vector<gvk::mesh_index_t> meshIndices;
+				meshIndices.push_back(meshIndex);
+				// Build meshlets:
+				std::tie(drawCallData.mPositions, drawCallData.mIndices) = gvk::get_vertices_and_indices(selection);
+				drawCallData.mNormals = gvk::get_normals(selection);
+				drawCallData.mTexCoords = gvk::get_2d_texture_coordinates(selection, 0);
+				// Get bone indices and weights
+				drawCallData.mBoneIndices = gvk::get_bone_indices_for_single_target_buffer(selection, meshIndicesInOrder);
+				drawCallData.mBoneWeights = gvk::get_bone_weights(selection);
+
+				// create selection for the meshlets
+				std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<gvk::mesh_index_t>>> meshletSelection;
+				meshletSelection.emplace_back(avk::shared(curModel), meshIndices);
+
+				auto cpuMeshlets = gvk::divide_into_meshlets(meshletSelection);
+#if USE_DIRECT_MESHLET
+#if USE_CACHE
+				gvk::serializer serializer("direct_meshlets-" + meshname + "-" + std::to_string(mpos) + ".cache");
+				auto [gpuMeshlets, _] = gvk::convert_for_gpu_usage_cached<gvk::meshlet_gpu_data<sNumVertices, sNumIndices>, sNumVertices, sNumIndices>(serializer, cpuMeshlets);
+#else
+				auto [gpuMeshlets, _] = gvk::convert_for_gpu_usage<gvk::meshlet_gpu_data<sNumVertices, sNumIndices>, sNumVertices, sNumIndices>(cpuMeshlets);
+#endif
+#else
+#if USE_CACHE
+				gvk::serializer serializer("indirect_meshlets-" + meshname + "-" + std::to_string(mpos) + ".cache");
+				auto [gpuMeshlets, generatedMeshletData] = gvk::convert_for_gpu_usage_cached<gvk::meshlet_redirected_gpu_data, sNumVertices, sNumIndices>(serializer, cpuMeshlets);
+#else
+				auto [gpuMeshlets, generatedMeshletData] = gvk::convert_for_gpu_usage<gvk::meshlet_redirected_gpu_data, sNumVertices, sNumIndices>(cpuMeshlets);
+#endif
+				drawCallData.mMeshletData = std::move(generatedMeshletData.value());
+#endif
+
+				// fill our own meshlets with the loaded/generated data
+				for (size_t mshltidx = 0; mshltidx < gpuMeshlets.size(); ++mshltidx) {
+					auto& genMeshlet = gpuMeshlets[mshltidx];
+
+					auto& ml = meshletsGeometry.emplace_back(meshlet{});
+
+#pragma region start to assemble meshlet struct
+					ml.mTransformationMatrix = drawCallData.mModelMatrix;
+					ml.mMaterialIndex = drawCallData.mMaterialIndex;
+					ml.mTexelBufferIndex = static_cast<uint32_t>(texelBufferIndex);
+					ml.mModelIndex = curEntry.mBoneMatricesBufferIndex;
+
+					ml.mGeometry = genMeshlet;
+#pragma endregion 
+				}
+			}
+		} // end for (auto& loadInfo : toLoad)
+
+		// create buffers for animation data
+		for (int i = 0; i < loadedModels.size(); ++i) {
```

Comment:
```suggestion
		for (size_t i = 0; i < loadedModels.size(); ++i) {
```

We strive to avoid warnings. ^^